src/pip/_internal/resolution/resolvelib/provider.py

```diff
@@ -242,9 +242,9 @@ def _eligible_for_upgrade(identifier: str) -> bool:
     def is_satisfied_by(self, requirement: Requirement, candidate: Candidate) -> bool:
         return requirement.is_satisfied_by(candidate)
 
-    def get_dependencies(self, candidate: Candidate) -> Sequence[Requirement]:
+    def get_dependencies(self, candidate: Candidate) -> Iterable[Requirement]:
         with_requires = not self._ignore_dependencies
-        return [r for r in candidate.iter_dependencies(with_requires) if r is not None]
+        return (r for r in candidate.iter_dependencies(with_requires) if r is not None)
```

Comment:
Why change to an iterator here? I’m not against doing so, but is it needed for this change? If it is, then a comment explaining why would be important.

OK, I think I see why -  if it’s not an iterator, we precompute the dependencies, which defeats the purpose of getting requires-python first. That definitely warrants a comment.